src/commands/commit/promote.ts

```diff
@@ -6,34 +6,66 @@ import { withSpinner } from "@/lib/helpers/request";
 import { promptToConfirm } from "@/lib/helpers/ux";
 
 export default class CommitPromote extends BaseCommand<typeof CommitPromote> {
-  static summary = "Promote all changes to the destination environment.";
+  static summary =
+    "Promote a single commit to the subsequent environment or all changes to the destination environment.";
 
   static flags = {
     to: Flags.string({
       summary:
-        "The destination environment to promote changes from the preceding environment.",
-      required: true,
+        "The destination environment to promote all changes from the preceding environment.",
     }),
     force: Flags.boolean({
       summary: "Remove the confirmation prompt.",
     }),
+    only: Flags.string({
+      summary: "The ID of the commit to promote to the subsequent environment",
+    }),
   };
 
   async run(): Promise<void> {
     const { flags } = this.props;
 
-    // Confirm first as we are about to promote changes to go live in the target
-    // environment, unless forced.
-    const prompt = `Promote all changes to \`${flags.to}\` environment?`;
-    const input = flags.force || (await promptToConfirm(prompt));
-    if (!input) return;
+    // Currently this command support two modes of operation based on the flags:
+    // * --to : Promotes all changes to the destination environment.
+    // * --only: Promotes one commit to the subsequent enviroment
+    // The absence or presence of both flags will result in an error.
+
```

Comment:
I've open this PR as a brief so we can confirm this is the expected behavior:

This command has now the two operation modes based on the flags:
* --to : Promotes all changes to the destination environment.
 * --only: Promotes one commit to the subsequent enviroment
This also means that the `--to` and `--only` flags can't be used at the same time, but at least one of them must be set.

The solution I found is to check the presence/absence of these here, and based on the flag received call `promoteAllChanges` or `promoteCommitChanges `

The good thing of this approach is that we can encapsulate both options (promoting one commit / promoting all changes) in one single command, but the downside is that we need to do this custom handle here.

Another option could be to separate this into two commands:
* knock commit promote-all (or only promote)
* knock commit promote-one